### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by MichaelStausholm**
Some minor changes + tests to improve test coverage.

There might be some old tests that are no longer needed / test already tested code.

### CODE REVIEW

File: core/src/main/java/dk/alexandra/fresco/lib/math/bool/log/Logarithm.java [hunk 1]
```diff
@@ -80,9 +80,6 @@ public Logarithm(List<DRes<SBool>> number) {
    * Computes the floor(log_2(x)).
    */
   private int log2(int n) {
-    if (n <= 0) {
```
Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/bool/log/Logarithm.java]: Why delete this check?

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java [hunk 2]
```diff
@@ -1,11 +1,11 @@
 package dk.alexandra.fresco.framework.util;
 
-import static org.junit.Assert.assertArrayEquals;
-import static org.junit.Assert.assertTrue;
 
+import org.junit.Test;
 import dk.alexandra.fresco.framework.MPCException;
+import static org.junit.Assert.assertArrayEquals;
```
Comment (GuutBoy) [inline core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java]: Static imports should be at the top

File: core/src/main/java/dk/alexandra/fresco/lib/math/bool/log/Logarithm.java (hunk 1 above)
Comment (MichaelStausholm) [reply core/src/main/java/dk/alexandra/fresco/lib/math/bool/log/Logarithm.java]: It's a private method and could never be invoked with n<=0

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java (hunk 2 above)
Comment (MichaelStausholm) [reply core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java]: Fixed.

File: core/src/main/java/dk/alexandra/fresco/lib/collections/sort/OddEvenMerge.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ private void merge(int first, int length, int step, ProtocolBuilderBinary builde
         }
         return null;
       });
-    } else if (length == 2) {
```
Comment (Rhaea) [inline core/src/main/java/dk/alexandra/fresco/lib/collections/sort/OddEvenMerge.java]: I would say either remove it entirely if it shouldn't be there, or have it stay. No point in having old incorrect code lying around in comments.

File: core/src/main/java/dk/alexandra/fresco/lib/compare/MiscBigIntegerGenerators.java [hunk 4]
```diff
@@ -90,9 +90,9 @@ public MiscBigIntegerGenerators(BigInteger modulus) {
 
     for (int i = 1; i <= l; i++) {
       int k = i;
-      if (i >= m) {
```
Comment (Rhaea) [inline core/src/main/java/dk/alexandra/fresco/lib/compare/MiscBigIntegerGenerators.java]: Again, if this is the case, just remove it. Can keep the actual comment though as that might be useful.

File: core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java [hunk 5]
```diff
@@ -111,8 +107,7 @@ private void parseLine(String line, ProtocolBuilderBinary builder) throws IOExce
     // System.out.println("IN: " + Arrays.toString(in));
     // System.out.println("OUT: " + Arrays.toString(out));
 
-    switch (type) {
```
Comment (Rhaea) [inline core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java]: Switch stmt makes it harder to do 100% testing? I see no other reason for changing this.

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestBinaryMatrix.java [hunk 6]
```diff
@@ -6,10 +6,14 @@
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertTrue;
 
+
 import java.util.ArrayList;
 import java.util.BitSet;
 import java.util.List;
 import java.util.Random;
+
```
Comment (Rhaea) [inline core/src/test/java/dk/alexandra/fresco/framework/util/TestBinaryMatrix.java]: I think the code style checker complains about this space :)

File: core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import dk.alexandra.fresco.framework.value.SBool;
+import dk.alexandra.fresco.framework.DRes;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class TestBinaryEquality {
```
Comment (Rhaea) [inline core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java]: Class name could probably be a bit better. What about TestBinaryEQInputLength? The other name indicates a different set of tests.

File: core/src/test/java/dk/alexandra/fresco/lib/math/bool/add/TestFullAdder.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import dk.alexandra.fresco.framework.value.SBool;
+import dk.alexandra.fresco.framework.DRes;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class TestFullAdder {
```
Comment (Rhaea) [inline core/src/test/java/dk/alexandra/fresco/lib/math/bool/add/TestFullAdder.java]: Same for this name. Not sure what the best approach is, but a longer class name is needed I guess

File: core/src/main/java/dk/alexandra/fresco/lib/collections/sort/OddEvenMerge.java (hunk 3 above)
Comment (GuutBoy) [reply core/src/main/java/dk/alexandra/fresco/lib/collections/sort/OddEvenMerge.java]: Agree, please dont leave code commented out. Either keep it or delete it. This is an issue a few other places

File: core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java (hunk 5 above)
Comment (GuutBoy) [reply core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java]: I don't really see the reason either. At least it should be `if ... else ...` i guess. Also, please delete all the commented code.

Comment (GuutBoy) [reply core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java]: ... also you could reduce the complexity of this code by having just a single return statement at the end. There is really no reason to have them in each case.

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java (hunk 2 above)
Comment (GuutBoy) [reply core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java]: Sorry, but static and non static should be in separate blocks and each block should be ordered alphabetically. https://google.github.io/styleguide/javaguide.html#s3.3-import-statements . It's really easier to do this organization automatically. Let me know if you need help setting up the IDE.

File: core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java (hunk 7 above)
Comment (MichaelStausholm) [reply core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java]: I don't agree. I believe the class name should reflect the class being tested and then the method names can indicate exactly what is being tested. This is the naming model we've used for the other tests as well, eg. "TestBitSetUtils", "TestCreditRater", etc.

File: core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java (hunk 5 above)
Comment (MichaelStausholm) [reply core/src/main/java/dk/alexandra/fresco/lib/helper/bristol/BristolCircuitParser.java]: The switch is hard to test (apparently it leaves out 3 branches somehow), but I have added else statements.

File: core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java (hunk 7 above)
Comment (GuutBoy) [reply core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java]: I agree with Michael. This way you can extend the test class with more tests on the same class without having to change the name.

Comment (Rhaea) [reply core/src/test/java/dk/alexandra/fresco/lib/compare/bool/eq/TestBinaryEquality.java]: Okay, makes sense. Ignore comment then.

**Pull request merged**